experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricExporter.ts

```diff
@@ -16,40 +16,38 @@
 
 import { AggregationTemporality } from './AggregationTemporality';
 import { MetricData } from './MetricData';
+import {
+  ExportResult,
+  ExportResultCode,
+} from '@opentelemetry/core';
 
 
 // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricexporter
 
-// TODO should this just be an interface and exporters can implement their own shutdown?
-export abstract class MetricExporter {
+export abstract class PushMetricExporter {
   protected _shutdown = false;
 
-  abstract export(batch: MetricData[]): Promise<void>;
+  abstract export(batch: MetricData[]): Promise<ExportResult>;
```

Comment:
Could we keep this `export` signature aligned with `SpanExporter.export`, the second parameter to be a callback of results?

Reply:
I remember following some discussion that concluded in favour promises but I might be wrong. I think being consistent is a good thing. I am fine with either way. What do others think?

Reply:
Maybe you're referring to this one: https://github.com/open-telemetry/opentelemetry-js/pull/2681#discussion_r774045108

> legendecas: IIUC, this might be mimicking the SpanExporter export case. However, I'm thinking that the SpanExporter is user implementable whilst the MetricReader and MeterProvider are not and we can provide a cleaner API with promise?